src/lib/logical_query_plan/abstract_lqp_node.hpp

```diff
@@ -146,11 +147,28 @@ class AbstractLQPNode : public std::enable_shared_from_this<AbstractLQPNode> {
    */
   ColumnID get_column_id(const AbstractExpression& expression) const;
 
+  /**
+   * Searches all output expressions.
+   * @return A shared pointer to the LQPColumnExpression with @param column_id or std::nullopt if it cannot be found.
+   */
+  std::optional<const std::shared_ptr<LQPColumnExpression>> find_column_expression(const ColumnID column_id) const;
+
   /**
    * @return whether the output column at @param column_id is nullable
    */
   virtual bool is_column_nullable(const ColumnID column_id) const;
 
+  /**
+   * @return A shared pointer to a vector of unique constraints valid for the current LQP.
+   * See lqp_unique_constraint.hpp for documentation. Nodes that query data must override this function.
+   */
+  [[nodiscard]] virtual std::shared_ptr<LQPUniqueConstraints> unique_constraints() const;
+
+  /**
+   * @return True, if a unique constraint exists for the given subset of output expressions. False otherwise.
+   */
+  [[nodiscard]] bool has_unique_constraint(const ExpressionUnorderedSet& output_expressions_subset) const;
```

Comment:
It would return false because [a, b] != [a]

The following is compatible with unique constraint [a, b], not with [a]
| a | b |
| - | - |
| **0** | 1 |
| **0** | **2** |
| 1 | **2** |
| 2 | 3 |

Reply:
However, the reverse case seems more interesting: Given an LQP constraint [a], what happens when calling has_unique_constraint([a,b])?

Well, currently, the function does what the doc says: It stupidly checks for an exact match.

But from a logical standpoint it could return true because [a] => [a, b], correct?

Reply:
That reverse case was actually what I was thinking about. We should make it more explicit if we mean `has_exact_unique_constraint` or `has_matching_unique_constraint`. In terms of helper methods, I feel like the second one is more useful.

Reply:
I like your attention to detail. I'm fine with `has_matching_unique_constraint` 👍